cloudant-client/src/main/java/com/cloudant/client/org/lightcouch/CouchDbClient.java

```diff
@@ -92,19 +89,24 @@
 
         this.gson = GsonHelper.initGson(new GsonBuilder()).create();
 
-        //if OkHttp is available then use it for connection pooling, otherwise default to the
-        //JVM built-in pooling for HttpUrlConnection
-        if (OkHttpClientHttpUrlConnectionFactory.isOkUsable() && props.getMaxConnections() > 0) {
+        // If OkHttp is available then use it for connection pooling, otherwise default to the
+        // JVM built-in pooling for HttpUrlConnection
+        if (OkHttpClientHttpUrlConnectionFactory.isOkUsable()) {
+            log.config("Using OkHttp");
             OkHttpClientHttpUrlConnectionFactory factory = new
                     OkHttpClientHttpUrlConnectionFactory();
-            //keep connections open for as long as possible, anything over 2.5 minutes will be
-            //longer than the server
-            ConnectionPool pool = new ConnectionPool(props.getMaxConnections(), TimeUnit.MINUTES
-                    .toMillis(3));
-            factory.getOkHttpClient().setConnectionPool(pool);
+            final int maxConns = props.getMaxConnections();
+            if (maxConns > 0) {
+                log.config("Setting max connections to " + maxConns);
+                //keep connections open for as long as possible, anything over 2.5 minutes will be
+                //longer than the server
+                ConnectionPool pool = new ConnectionPool(maxConns, TimeUnit.MINUTES.toMillis(3));

```

Comment:
The point was that 3 > 2.5, but I've made the comment a bit more clear in bd1c0be.